tasks/ThrusterTask.cpp

```diff
@@ -0,0 +1,99 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ThrusterTask.hpp"
+
+#include <gazebo/msgs/msgs.hh>
+
+using namespace gazebo;
+using namespace rock_gazebo;
+
+ThrusterTask::ThrusterTask(std::string const& name)
+    : ThrusterTaskBase(name)
+{
+}
+
+ThrusterTask::ThrusterTask(std::string const& name, RTT::ExecutionEngine* engine)
+    : ThrusterTaskBase(name, engine)
+{
+}
+
+ThrusterTask::~ThrusterTask()
+{
+}
+
+bool ThrusterTask::configureHook()
+{
+    if (! ThrusterTaskBase::configureHook())
+        return false;
+
+    // Set gazebo topic to advertise
+    node = transport::NodePtr( new transport::Node() );
```

Comment:
It would make sense to delete the node in cleanupHook. Otherwise, a deconfigured task would look active (an use ressources) in the gazebo network.

Actually, you should probably
 - create and delete the node in resp. configureHook and cleanupHook
 - advertise and remove it from the network in resp. startHook and stopHook